libminifi/include/core/Annotation.h

```diff
@@ -0,0 +1,77 @@
+/**
+
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace core {
+
+template <typename T>
+struct AnnotationBase {
```

Comment:
are more kinds of annotations in the works? if so, we could drop the default definition, and specialize on `Annotation<EInputRequirement>` thus we wouldn't need the `InputRequirementAnnotationBase` anymore

Reply:
It's possible that we will have more annotations in the future. We can get rid of AnnotationBase, but not InputRequirementAnnotationBase, as a non-template base class in necessary for the dynamic cast in Processor::getInputRequirement().

Reply:
is `dynamic_cast<const Annotation<EInputRequirement>*>(_)` malformed?

Reply:
Okay, I understand now, it's done.